src/interactive-window/debugger/jupyter/kernelDebugAdapter.ts

```diff
@@ -82,14 +90,121 @@ export class KernelDebugAdapter extends KernelDebugAdapterBase {
                     metadata.interactive.lineIndex +
                     (metadata.generatedCode?.lineOffsetRelativeToIndexOfFirstLineInCell || 0)
             });
-            this.cellToFile.set(Uri.parse(metadata.interactive.uristring), {
-                path: norm,
+            this.cellToDebugFileSortedInReverseOrderByLineNumber.push({
+                debugFilePath: norm,
+                interactiveWindow: Uri.parse(metadata.interactive.uristring),
+                metadata,
                 lineOffset:
                     metadata.interactive.lineIndex +
                     (metadata.generatedCode?.lineOffsetRelativeToIndexOfFirstLineInCell || 0)
             });
+            // Order cells in reverse order.
+            this.cellToDebugFileSortedInReverseOrderByLineNumber.sort(
+                (a, b) => b.metadata.interactive.lineIndex - a.metadata.interactive.lineIndex
+            );
         } catch (err) {
             traceError(`Failed to dump cell for ${cell.index} with code ${metadata.interactive.originalSource}`, err);
         }
     }
+    protected override translateDebuggerFileToRealFile(
+        source: DebugProtocol.Source | undefined,
+        lines?: { line?: number; endLine?: number; lines?: number[] }
+    ) {
+        if (!source || !source.path || !lines || (typeof lines.line !== 'number' && !Array.isArray(lines.lines))) {
+            return;
+        }
+        // Find the cell that matches this line in the IW file.
+        const cell = this.cellToDebugFileSortedInReverseOrderByLineNumber.find(
+            (item) => item.debugFilePath === source.path
```

Comment:
Comment seems wrong here. It's not searching by line but rather by the translated debugFilePath. That does map to a specific cell entry.